From 5dbb49abc08acb9a484bc187c18c433fb821d281 Mon Sep 17 00:00:00 2001
From: Yi Fan Yu <yifan.yu@windriver.com>
Date: Wed, 17 Mar 2021 12:16:31 -0400
Subject: [PATCH] remove s390x_vec_op_t

s390x_vec_op_t is not needed anywhere, only elements of enum are accessed
removing it ensures that valgrind can be built with -fno-common option

Fixes
ld: ../../VEX/libvex-amd64-linux.a(libvex_amd64_linux_a-guest_s390_helpers.o):/usr/src/debug/valgrind/3.15.0-r0/build/VEX/../../valgrind-3.15.0/VEX/priv/guest_s390_defs.h:289: multiple definition of `s390x_vec_op_t'; ../../VEX/libvexmultiarch-amd64-linux.a(libvexmultiarch_amd64_linux_a-multiarch_main_main.o):/usr/src/debug/valgrind/3.15.0-r0/build/VEX/../../valgrind-3.15.0/VEX/priv/guest_s390_defs.h:289: first defined here

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Signed-off-by: Yi Fan Yu <yifan.yu@windriver.com>
---
 VEX/priv/guest_s390_defs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/VEX/priv/guest_s390_defs.h b/VEX/priv/guest_s390_defs.h
index a64d563..fb963e9 100644
--- a/VEX/priv/guest_s390_defs.h
+++ b/VEX/priv/guest_s390_defs.h
@@ -286,7 +286,7 @@ typedef enum {
    S390_VEC_OP_VCLFNL,
    S390_VEC_OP_VCRNF,
    S390_VEC_OP_LAST             // supposed to be the last element in enum
-} s390x_vec_op_t;
+};
 
 /* Arguments of s390x_dirtyhelper_vec_op(...) which are packed into one
    ULong variable.
